pubsub/gcp/gcp.go

```diff
@@ -133,15 +130,26 @@ var _ pubsub.Publisher = &publisher{}
 var _ pubsub.MultiPublisher = &publisher{}
 
 // NewPublisher will instantiate a new GCP MultiPublisher.
-func NewPublisher(ctx context.Context, projID, topic string, opts ...option.ClientOption) (pubsub.MultiPublisher, error) {
-	client, err := gpubsub.NewClient(ctx, projID, opts...)
+func NewPublisher(ctx context.Context, cfg Config) (pubsub.MultiPublisher, error) {
+	p := &publisher{}
+
+	if cfg.ProjectID == "" {
+		return p, errors.New("project id is required")
+	}
+	if cfg.Topic == "" {
+		return p, errors.New("topic name is required")
+	}
+	if cfg.Subscription == "" {
```

Comment:
Subscriptions are not required to create a new publisher. We can remove this check.